src/prefect/transactions.py

```diff
@@ -218,6 +236,60 @@ def transaction(
     commit_mode: CommitMode = CommitMode.LAZY,
     overwrite: bool = False,
 ) -> Generator[Transaction, None, None]:
+    """
+    A context manager for opening and managing a transaction.
+
+    Args:
+        - key: An identifier to use for the transaction
+        - store: The store to use for persisting the transaction result. If not provided,
+            a default store will be used based on the current run context.
+        - commit_mode: The commit mode controlling when the transaction and
+            child transactions are committed
+        - overwrite: Whether to overwrite an existing transaction record in the store
+
+    Yields:
+        - Transaction: An object representing the transaction state
+    """
+    # if there is no key, we won't persist a record
+    if key and not store:
+        flow_run_context = FlowRunContext.get()
+        task_run_context = TaskRunContext.get()
+        existing_factory = getattr(task_run_context, "result_factory", None) or getattr(
+            flow_run_context, "result_factory", None
+        )
+
+        if existing_factory and existing_factory.storage_block_id:
+            new_factory = existing_factory.model_copy(
+                update={
+                    "persist_result": True,
+                }
+            )
+        else:
+            default_storage = get_default_result_storage(_sync=True)
+            if not default_storage._block_document_id:
+                default_name = PREFECT_DEFAULT_RESULT_STORAGE_BLOCK.value().split("/")[
+                    -1
+                ]
+                default_storage.save(default_name, _sync=True)
```

Comment:
does this need an overwrite flag?

Reply:
It shouldn't because we only run this save after confirming it hasn't been loaded from the server, but I'll add `overwrite=True` to be safe since that should guard against race conditions.